### DISCUSSION TITLE
Weird quoted PATH problem with ubuntu

### DISCUSSION THREAD

**Discussion by mulle-nat** [nektos/act#1511] [General]
When I run a very [simple workflow](https://github.com/mulle-nat/act-path/blob/master/.github/workflows/ci.yml), that just outputs the `PATH` variable, I see a difference between "bullseye" and "jammy", namely that for "jammy" the PATH is somewhat corrupted with a quoted PATH:

## Debian works

``` bash
$ docker run -t debian:bullseye-slim bash -c 'echo $PATH'
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
$ act -P ubuntu-latest=debian:bullseye-slim
[CI/build]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/0] user= workdir=
| PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

## Ubuntu doesn't

``` bash
$ docker run -t ubuntu:jammy bash -c 'echo $PATH'
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
$ act -P ubuntu-latest=ubuntu:jammy
[CI/build]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/0] user= workdir=
| PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin":/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

I think it must be an `act` bug somewhere.

**Comment by ChristopherHX** [comment]
I think this is something, which https://github.com/nektos/act/pull/1503 would fix.

It is possible that ubuntu's /etc/environment file contains a line such as `PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"` and the parser doesn't support quotes.

**Comment by mulle-nat** [comment]
There is a distinction, in that bullseye defines PATH in `/etc/profile`:

``` bash
$ docker run -t debian:bullseye-slim bash -c 'grep PATH /etc/profile'
  PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
  PATH="/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games"
export PATH
```

and  ubuntu indeed defines PATH in `/etc/environment`:

``` bash
$ docker run -t ubuntu:jammy bash -c 'grep PATH /etc/environment'
PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
```

But the quoting looks the same.

**Reply by ChristopherHX** [reply]
Thanks for confirming, act doesn't read `/etc/profile` and uses the hardcoded default `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin` on debian.

Can you test the ubuntu and debian image with the artifact of https://github.com/nektos/act/actions/runs/3688632015 it's a build of https://github.com/nektos/act/pull/1503? Then you should not see weird PATH values.

**Comment by mulle-nat** [comment]
Creates this output

```
$ ~/Downloads/act -P ubuntu-latest=ubuntu:jammy
[CI/build] 🚀  Start image=ubuntu:jammy
[CI/build]   🐳  docker pull image=ubuntu:jammy platform= username= forcePull=false
[CI/build]   🐳  docker create image=ubuntu:jammy platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[CI/build]   🐳  docker run image=ubuntu:jammy platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[CI/build] 🧪  Matrix: map[os:ubuntu-latest]
[CI/build] ⭐ Run Main Dump Environment
[CI/build]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/0] user= workdir=
| PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
[CI/build]   ✅  Success - Main Dump Environment
[CI/build] 🏁  Job succeeded
```

This doesn't have the quoting problem, but it just seems to ignore `/etc/environment`.

**Reply by ChristopherHX** [reply]
actions/runner does the same thing for job container.

A `env PATH="..."` of a Dockerfile should be respected like in actions/runner.

**Comment by mulle-nat** [comment]
If thats more compatible with the way github works, than its better IMO :smile:
